lib/widgets/components/purchase/purchase_overlay.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:coffeecard/base/style/colors.dart';
+import 'package:coffeecard/cubits/purchase/purchase_cubit.dart';
+import 'package:coffeecard/cubits/receipt/receipt_cubit.dart';
+import 'package:coffeecard/cubits/tickets_page/tickets_cubit.dart';
+import 'package:coffeecard/models/receipts/receipt.dart';
+import 'package:coffeecard/models/ticket/product.dart';
+import 'package:coffeecard/payment/payment_handler.dart';
+import 'package:coffeecard/service_locator.dart';
+import 'package:coffeecard/widgets/components/purchase/purchase_process.dart';
+import 'package:coffeecard/widgets/components/receipt/receipt_overlay.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class PurchaseOverlay {
+  final BuildContext _context;
+
+  void hide() {
+    Navigator.of(_context, rootNavigator: true).pop();
+  }
+
+  void show(InternalPaymentType paymentType, Product product) {
+    showDialog(
+      context: _context,
+      barrierColor: AppColor.scrim,
+      barrierDismissible: false,
+      useRootNavigator: true,
+      builder: (_) {
+        return WillPopScope(
+          // Will prevent Android back button from closing overlay.
+          onWillPop: () async => false,
+          child: BlocProvider(
+            create: (context) => PurchaseCubit(
+              paymentHandler: PaymentHandler(paymentType),
+              product: product,
+            ),
+            child: BlocListener<PurchaseCubit, PurchaseState>(
+              listener: (context, state) async {
+                if (state is PurchaseCompleted) {
+                  hide(); //Removes this overlay
+                  final navigator = Navigator.of(context);
+                  //TODO consider using popUntil instead of this
+                  while (navigator.canPop()) {
+                    //Gets the user back to the home-screen of the app
+                    navigator.pop();
+                  }
+                  //TODO Consider if these calls should be moved elsewhere, e.g. inside the purchase cubit
```

Comment:
Personally, I don't really think the calls to refresh the tickets, and receipts belong here, but not sure where else they make sense to have. Thoughts? If we keep them here, we should move the cubits to the constructor for testability, at the very least.

Reply:
I think this is fine for now. We should probably do something smarter once we get to testing our logic but for now i would prefer to get this into dev :)

Reply:
Alright, in that case I will just leave the TODO for now, so we can get back to it later